### PR TITLE
[AKS] Support BYO control plane identity

### PR DESCRIPTION
**Pull request by norshtein**
**Description**  

The feature "bring-your-own control plane identity" is going to GA. This PR adds cli support for it.
**Testing Guide**  

Added unit test and e2e recording.
**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls help to review? thanks

File: src/azure-cli/azure/cli/command_modules/acs/_params.py [hunk 1]
```diff
@@ -208,6 +208,7 @@ def load_arguments(self, _):
         c.argument('enable_private_cluster', action='store_true')
         c.argument('nodepool_tags', nargs='*', validator=validate_nodepool_tags, help='space-separated tags: key[=value] [key[=value] ...]. Use "" to clear existing tags.')
         c.argument('enable_managed_identity', action='store_true')
+        c.argument('assign_identity', type=str, validator=validate_assign_identity)
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/acs/_params.py:211]: Is there a way to remove identity?

File: src/azure-cli/azure/cli/command_modules/acs/_validators.py [hunk 2]
```diff
@@ -392,3 +392,12 @@ def validate_label(label):
...
     return {kv[0]: kv[1]}
+
+
+def validate_assign_identity(namespace):
+    if namespace.assign_identity is not None:
+        if namespace.assign_identity == '':
+            return
+        from msrestazure.tools import is_valid_resource_id
+        if not is_valid_resource_id(namespace.assign_identity):
+            raise CLIError("--assign-identity is not a valid Azure resource ID.")
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/acs/_validators.py:403]: Can you use [`InvalidArgumentValueError`](https://github.com/Azure/azure-cli/blob/7b927fe3a01e25377571532ec22c48425bdc8a62/src/azure-cli-core/azure/cli/core/azclierror.py#L123) here?

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 3]
```diff
@@ -756,6 +758,25 @@ def _generate_properties(api_version, orchestrator_type, orchestrator_version, m
...
+    match = re.search(pattern, resource_id)
+    if match:
+        resource_group_name = match.group(1)
+        identity_name = match.group(2)
+        try:
+            identity = msi_client.user_assigned_identities.get(resource_group_name=resource_group_name,
+                                                               resource_name=identity_name)
+        except CloudError as ex:
+            if 'was not found' in ex.message:
+                raise CLIError("Identity {} not found.".format(resource_id))
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:774]: Can you use [`ResourceNotFoundError`](https://github.com/Azure/azure-cli/blob/7b927fe3a01e25377571532ec22c48425bdc8a62/src/azure-cli-core/azure/cli/core/azclierror.py#L151) here?

File: src/azure-cli/azure/cli/command_modules/acs/_params.py (hunk 1 above)
Comment (norshtein) [reply src/azure-cli/azure/cli/command_modules/acs/_params.py:211]: There is a preview feature which can switch the identity, but at least one and only one identity should be associated with the cluster.

**Comment by norshtein** [comment]
Hi @fengzhou-msft , thanks for reviewing! I've updated error type, could you please have a look now?

**Review by fengzhou-msft — APPROVED** [review]

**Review by arrownj — APPROVED** [review]
LGTM

**Comment by arrownj** [comment]
Hi @norshtein , recently we have concluded an [error handling guide](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md). Maybe you can have a look and try to follow it in the future. In later PRs, we will reject to raise `CLIError`. It's better to update this PR's `raise CLIError` part. Considering the time frame, I'm also OK for keeping it for this PR. Please kindly be noted. Thanks

**Comment by norshtein** [comment]
@arrownj , thanks for the explanation, I just tried to update all `CLIError` added in this PR to correct category according to the manual.

**Comment by norshtein** [comment]
Close and reopen to rerun CI.

**Comment by norshtein** [comment]
The CI failed, seems to be a transient error, could you please help rerun it @arrownj @fengzhou-msft

**Pull request merged**